cjwkernel/types.py

```diff
@@ -634,10 +644,10 @@ def trans(
 
     @classmethod
     def from_dict(cls, value: Dict[str, Any]) -> I18nMessage:
-        return cls(value["id"], value["arguments"])
+        return cls(value["id"], value["arguments"], value["source"])
 
     def to_dict(self) -> Dict[str, Any]:
-        return {"id": self.id, "arguments": self.args}
+        return {"id": self.id, "arguments": self.args, "source": self.source}
```

Comment:
To stay consistent with the thousands of I18nMessages already in our database, don't produce a `source` key if `self.source is None`.